tests/unit/auth/v2/headerAuthCheck.js

```diff
@@ -0,0 +1,40 @@
+const assert = require('assert');
+
+const headerAuthCheck =
+    require('../../../../lib/auth/v2/headerAuthCheck').check;
+const DummyRequestLogger = require('../../helpers').DummyRequestLogger;
+
+const log = new DummyRequestLogger();
+
+describe('v2: headerAuthCheck', () => {
+    it('should not fail if the token is missing', () => {
+        const request = {
+            headers: {},
+        };
+        const res = headerAuthCheck(request, log, {});
+        assert.notStrictEqual(res.err, undefined);
+        assert.notStrictEqual(res.err.InvalidToken, true);
+    });
+
+    it('should fail if the token is invalid', () => {
+        const request = {
+            headers: {
+                'x-amz-security-token': 'invalid-token',
+            },
+        };
+        const res = headerAuthCheck(request, log, {});
+        assert.notStrictEqual(res.err, undefined);
+        assert.strictEqual(res.err.InvalidToken, true);
+    });
+
+    it('should succeed if the token is valid', () => {
```

Comment:
the notion of valid / invalid token will not be manage by arsenal, from here a token is valid when the syntaxe is correct :)